src/main/java/org/apache/mesos/state/SchedulerState.java

```diff
@@ -0,0 +1,154 @@
+package org.apache.mesos.state;
+
+import com.google.protobuf.TextFormat;
+import org.apache.mesos.Protos.*;
+import org.apache.mesos.curator.CuratorStateStore;
+import org.apache.mesos.offer.TaskException;
+import org.apache.mesos.offer.TaskUtils;
+import org.apache.mesos.reconciliation.TaskStatusProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ * Read/write interface for the state of a Scheduler.
+ */
+public class SchedulerState extends CuratorStateStore implements TaskStatusProvider {
+    private static final Logger log = LoggerFactory.getLogger(SchedulerState.class);
+
+    private static final String SUPPRESSED_KEY = "suppressed";
+
+    public SchedulerState(String frameworkName, String mesosZkURI) {
+        super(frameworkName, mesosZkURI);
+    }
+
+    public FrameworkID getFrameworkId() {
+        try {
+            return fetchFrameworkId().orElse(null);
```

Comment:
We just went through a series of changes to avoid returning nulls.  Let's not transform [the underlying `StateStore` interface](https://github.com/mesosphere/dcos-commons/blob/6a5e665c56dcb9eb478b89619453f4f5e0fdbad2/src/main/java/org/apache/mesos/state/StateStore.java#L53) here.